source/proof_document/spec/proof_document_spec.rb

```diff
@@ -0,0 +1,233 @@
+require 'securerandom'
+require 'identity-idp-functions/proof_document'
+require 'identity_doc_auth'
+
+RSpec.describe IdentityIdpFunctions::ProofDocument do
+  let(:idp_api_auth_token) { SecureRandom.hex }
+  let(:callback_url) { 'https://example.login.gov/api/callbacks/proof-document/:token' }
+  let(:event) do
+    {
+      encryption_key: '12345678901234567890123456789012',
+      front_image_iv: '1234567890123456',
+      back_image_iv: '1234567890123456',
+      selfie_image_iv: '1234567890123456',
+      front_image_url: 'http://foo.com/bar1',
+      back_image_url: 'http://foo.com/bar2',
+      selfie_image_url: 'http://foo.com/bar3',
+      liveness_checking_enabled: true,
+      callback_url: callback_url,
+    }
+  end
+  let(:applicant_pii) do
+    {
+      first_name: 'Johnny',
+      last_name: 'Appleseed',
+      uuid: SecureRandom.hex,
+      dob: '01/01/1970',
+      ssn: '123456789',
+      phone: '18888675309',
+    }
+  end
+
+  before do
+    stub_const(
+      'ENV',
+      'IDP_API_AUTH_TOKEN' => idp_api_auth_token,
+      'acuant_assure_id_password' => 'aaa',
+      'acuant_assure_id_subscription_id' => 'aaa',
+      'acuant_assure_id_url' => 'https://example.com',
+      'acuant_assure_id_username' => 'aaa',
+      'acuant_facial_match_url' => 'https://facial_match.example.com',
+      'acuant_passlive_url' => 'https://liveness.example.com',
+      'acuant_timeout' => 60,
+    )
+
+    stub_request(:get, 'http://169.254.169.254/2016-09-02/dynamic/instance-identity/document').
+        to_return(body: {
+            'region' => 'us-west-1',
+            'accountId' => '12345',
+        }.to_json)
+
+    url = URI.join('https://example.com', '/AssureIDService/Document/Instance')
+    stub_request(:post, url).to_return(body: '"this-is-a-test-instance-id"')
+    stub_request(:post, "https://example.com/AssureIDService/Document/this-is-a-test-instance-id/Image?light=0&side=0").to_return(body: '')
+    stub_request(:post, "https://example.com/AssureIDService/Document/this-is-a-test-instance-id/Image?light=0&side=1").to_return(body: '')
+    stub_request(:get, "https://example.com/AssureIDService/Document/this-is-a-test-instance-id").to_return(body: '{"Result":1}')
+    stub_request(:get, "https://example.com/AssureIDService/Document/this-is-a-test-instance-id/Field/Image?key=Photo").to_return(body: '')
+    stub_request(:post, "https://facial_match.example.com/api/v1/facematch").to_return(body:'{"IsMatch":true}')
+    stub_request(:post, "https://liveness.example.com/api/v1/liveness").to_return(body:'{"LivenessResult":{"LivenessAssessment": "Live"}}')
+
+    allow_any_instance_of(OpenSSL::Cipher::AES).to receive(:update).and_return('foo')
+    allow_any_instance_of(OpenSSL::Cipher::AES).to receive(:final).and_return('bar')
+    allow_any_instance_of(IdentityDocAuth::Acuant::Responses::GetResultsResponse).to receive(:pii_from_doc).and_return(applicant_pii)
```

Comment:
We have the mock proofer class in the gem so we don't need to stub any instance like this, we can use `IdentityDocAuth::Mock::DocAuthMockClient.mock_response`